xivo_dao/agent_status_dao.py

```diff
@@ -57,6 +67,17 @@ def _get_login_status_by_number(session, agent_number, tenant_uuids=None):
     return login_status.first()
 
 
+def _get_login_status_by_user(session, user_uuid, tenant_uuids=None):
+    login_status = (session
+                    .query(AgentLoginStatus)
+                    .outerjoin((AgentFeatures, AgentFeatures.id == AgentLoginStatus.agent_id))
+                    .join((UserFeatures, AgentFeatures.id == UserFeatures.agentid))
+                    .filter(UserFeatures.uuid == user_uuid))
+    if tenant_uuids is not None:
+        login_status = login_status.filter(AgentFeatures.tenant_uuid.in_(tenant_uuids))
+    return login_status.first()
```

Comment:
If tenant_uuids is an empty list, you'll get a SQLAlchemy warning

Reply:
It's only valid for sqlalchemy==1.0.15 (stretch), but no more with buster version (1.2.18)